### PR TITLE
Add params to task node's metadata

### PR DESCRIPTION
**Pull request by limdauto**
## Description

This PR exposes a `parameters` field to a node's metadata. It contains the names and values of the `parameters` inputs to a node.

## Checklist

- [x] Read the [contributing](/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added new entries to the `RELEASE.md` file
- [ ] Added tests to cover my changes

## Legal notice

- [x] I acknowledge and agree that, by checking this box and clicking "Submit Pull Request":

- I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
- I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorised to submit this contribution on behalf of the original creator(s) or their licensees.
- I certify that the use of this contribution as authorised by the Apache 2.0 license does not violate the intellectual property rights of anyone else.

### CODE REVIEW

**Comment by studioswong** [comment]
Thanks for the work @limdauto ! For clarity and documentation purpose, can you please provide an example metadata object returned from this new change in the description so everyone on the viz team will be aware of how the new `task` node metadata will look like? 

Also, just to double confirm ( and please definitely correct me if I'm wrong), the `params`/`parameters`(?) field will be returned from the metadata regardless whether there are any? 

It looks like some pylint tests are failing on the CI - not sure if it's related to these changes but please check?

All in all, thanks for exposing the field to the metadata 👍

**Comment by limdauto** [comment]
@studioswong for formal documentation, I will add it to the API documentation ticket. In the meantime, you can look at the test for an example:

```
assert data["code"] == inspect.getsource(salmon)
assert data["filepath"] == str(Path(project_root) / filepath)
assert data["docstring"] == inspect.getdoc(salmon)
assert data["parameters"] == {"name": "value"}
```

The `parameters` field will be a field on par with other `task` node's metadata. It won't appear when there is no parameters to display. There is no `params` key exposed to the frontend. `parameters` is the only key available.

File: package/kedro_viz/server.py [hunk 1]
```diff
@@ -351,7 +351,12 @@ def format_pipelines_data(pipelines: Dict[str, "Pipeline"]) -> Dict[str, list]:
     }
 
 
-# pylint: disable=too-many-locals,too-many-arguments
+def _is_namespace_param(namespace: str) -> bool:
+    """Returns whether a dataset namespace is a parameter"""
+    return "param" in namespace.lower()
```
Comment (lorenabalan) [inline package/kedro_viz/server.py:356]: I know this is what it was doing before (line 440), but I think technically `namespace.lower().startswith("param")` is more correct.

**Review by lorenabalan — APPROVED** [review]
Lgtm

**Pull request merged**